### ISSUE TITLE
MATLAB crashes trying to read file without permissions

### ISSUE THREAD

**Issue by cako** [ahay/src#116]
Executing `rsf_read_all` on a file without read permissions causes MATLAB to exit unexpectedly. The correct behavior is to raise an error.

```
$ matlab
matlab: build/api/c/file.c: Cannot read input (header) file file.rsf: Permission denied
Aborted
```

**Comment by henryk-modzelewski** [comment]
You are absolutely right. However, from what I remember the C API does not create recoverable error, hence one cannot delegate error handling in this case to MATLAB. I might be missing something, but after brief inspection I do not believe that anything changed since I wrote that code.
